### PR TITLE
fix: allow OSX interactive session during det user create

### PR DESCRIPTION
**Pull request by hkang1**
terminal output on OSX
```
Waiting for master to be available...
Waiting for master to be available...
Waiting for master to be available...
Starting determined_integrations_5300-agent-0
+ python /Users/ck/Projects/determined/webui/tests/bin/createUserAndExperiments.py
Preparing files (/Users/ck/Projects/determined/tests/integrations/fixtures/no_op) to send to master... 17.0KB and 16 files
Created experiment 1
Preparing files (/Users/ck/Projects/determined/tests/integrations/fixtures/no_op) to send to master... 17.0KB and 16 files
Created experiment 2
Preparing files (/Users/ck/Projects/determined/tests/integrations/fixtures/no_op) to send to master... 17.0KB and 16 files
Created experiment 3
Preparing files (/Users/ck/Projects/determined/tests/integrations/fixtures/no_op) to send to master... 17.0KB and 16 files
Created experiment 4
+ yarn --cwd /Users/ck/Projects/determined/webui/tests run cypress run --config-file cypress.json --config defaultCommandTimeout=4000,baseUrl=http://localhost:5300 --browser chrome --headless
yarn run v1.22.4
```

### CODE REVIEW

File: webui/tests/requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+pexpect
```
Comment (hamidzr) [inline webui/tests/requirements.txt:1]: suggestion: we are also using the docker package in the WebUI tests module can we add that as well?

File: webui/tests/bin/createUserAndExperiments.py [hunk 2]
```diff
@@ -26,7 +27,17 @@
 # but the `setsid` command does not seem to be generally available on macOS.)
 # FIXME The script is blocking here on macOS. We disable this and its corresponding tests for the
 # time being until it's fixed.
-# p = pexpect.spawn("det", ["-u", "admin", "u", "create", "hoid"])
-# p.expect("Password.*:")
-# p.sendline("")
-# p.wait()
+p = pexpect.spawn("det", ["-u", "admin", "u", "create", "hoid"])
+p.expect("Password.*:")
```
Comment (hamidzr) [inline webui/tests/bin/createUserAndExperiments.py:31]: seems like the `det u create` behavior has changed. it no longer expects you to enter a password. at least for passwordless account.

you would instead pair it with `det -u admin user change-password abc` to change the password which asks for it twice.

Comment (hamidzr) [reply webui/tests/bin/createUserAndExperiments.py:31]: wondering how it passed the tests ..

Comment (hamidzr) [reply webui/tests/bin/createUserAndExperiments.py:31]: it'd be great to add an accompanying e2e tests that logs in with the newly created user and a present non-zero passowrd

Comment (hkang1) [reply webui/tests/bin/createUserAndExperiments.py:44]: good call, updated to create a new user with a password.

File: webui/tests/bin/createUserAndExperiments.py [hunk 3]
```diff
@@ -26,7 +27,17 @@
...
+p.sendline("")
+
+# Avoid Hang on macOS
+# https://github.com/pytest-dev/pytest/issues/2022
+while True:
+    try:
+        p.read_nonblocking()
+    except Exception:
```
[the remark this answers is not in the document]
Comment (hkang1) [reply webui/tests/bin/createUserAndExperiments.py]: nope, this adds the waiting method that works on the OSX to avoid manual user entry

File: webui/tests/cypress/support/commands.js [hunk 4]
```diff
@@ -3,7 +3,16 @@
 // https://on.cypress.io/custom-commands
 // ***********************************************
 
-const DEFAULT_TEST_USER = 'determined';
+const DEFAULT_TEST_USERNAME = 'userwithpw';
+const DEFAULT_TEST_PASSWORD = 'specialpw';
+
+const sha512 = require('js-sha512').sha512;
+
+const saltAndHashPassword = (password) => {
```
[the remark this answers is not in the document]
Comment (hkang1) [reply webui/tests/cypress/support/commands.js:11]: I don't like duplicating the code base either. As you pointed out, having `tests` have a dependency on `react` seems more problematic to me in the long run.

**Comment by hkang1** [comment]
> suggestion: add a test that uses the UI to login with the user with a password while also keeping the user with no password test around so we test both paths :)

Great idea, I'll create a separate ticket for this as it's going to be a bit more work to sort that out.
